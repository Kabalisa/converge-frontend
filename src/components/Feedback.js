/* eslint react/no-array-index-key: 0 */
import React from 'react';
import PropTypes from 'prop-types';
import Slider from './commons/slider';

const Feedback = props => (
  props.feedback.map(({
    question,
    startDate,
    duration,
    responses,
    type,
    status,
  }, index) => (
    <tr key={index}>
      <td>{question}</td>
      <td>{type}</td>
      <td>{responses}</td>
      <td>{startDate}</td>
      <td>{duration}</td>
      <td>
        <img src="/1982b0427a5a73aeffde1f16255b4f8f.svg" alt="Edit" />
        <img src="/d34784bf1f4cacf5f9970140c723e361.svg" alt="Delete" />
      </td>
      <td><Slider checked={status} /></td>
    </tr>
  ))
);

Feedback.propTypes = {
  feedback: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Feedback;
