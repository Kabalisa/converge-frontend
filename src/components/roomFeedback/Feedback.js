/* eslint react/no-array-index-key: 0 */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import CheckBoxSlide from '../commons/CheckboxSlide';
import EditFeedbackComponent from './EditFeedback';
import { UPDATE_QUESTION_STATUS_MUTATION } from '../../graphql/mutations/Question';
import GET_ROOM_FEEDBACK_QUESTIONS_QUERY from '../../../src/graphql/queries/questions';
import DeleteFeedback from './DeleteFeedback'; // eslint-disable-line

/**
 * Feedback Component
 *
 * @param {array} props
 *
 * @returns {JSX}
 */
const Feedback = (props) => {
  const { role } = props;
  return (
    props.feedback.map(({
      id, question, startDate, endDate, questionTitle,
      questionResponseCount, questionType, isActive,
    }) => (
      <div className="table__row" key={id}>
        <span>
          {question}
        </span>
        <span>{questionType}</span>
        <span>{questionResponseCount}</span>
        <span>{props.startDateFormatter(startDate)}</span>
        <span>{props.durationFormatter(startDate, endDate)}</span>
        {
          role === '2' &&
          <Fragment>
            <span>
              {!isActive &&
                <EditFeedbackComponent
                  id="edit-modal"
                  questionId={id}
                  question={question}
                  questionType={questionType}
                  questionTitle={questionTitle}
                  startDate={startDate}
                  endDate={endDate}
                />}
              <DeleteFeedback id={id} question={question} />
            </span>
            <span className="checkbox">
              <CheckBoxSlide
                questionId={parseInt(id, 10)}
                isChecked={isActive}
                actionToPerform="mutation"
                mutationQuery={UPDATE_QUESTION_STATUS_MUTATION}
                updateQuery={GET_ROOM_FEEDBACK_QUESTIONS_QUERY}
              />
            </span>
          </Fragment>
        }
      </div>
    )));
};

Feedback.defaultProps = {
  feedback: [
    {
      question: 'There is no question so far',
      questionType: 'Input',
      startDate: '2019-02-21 23:42:43',
      endDate: '2019-02-21 23:42:43',
      questionResponseCount: 0,
      isActive: false,
    },
  ],
};

Feedback.propTypes = {
  feedback: PropTypes.arrayOf(PropTypes.object).isRequired,
  role: PropTypes.string.isRequired,
};

export default Feedback;
