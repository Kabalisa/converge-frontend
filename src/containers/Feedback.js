import React, { Fragment } from 'react';
import { NavBar } from '../components';
import RoomFeedback from '../components/RoomFeedback';

const Feedback = () => (
  <Fragment>
    <NavBar />
    <RoomFeedback />
  </Fragment>
);

export default Feedback;
