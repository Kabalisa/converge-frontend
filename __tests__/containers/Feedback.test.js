import React from 'react';
import { shallow } from 'enzyme';
import Feedback from '../../src/containers/Feedback';

describe('Feedback component', () => {
  const wrapper = shallow(<Feedback />);

  it('renders correctly in memory', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
