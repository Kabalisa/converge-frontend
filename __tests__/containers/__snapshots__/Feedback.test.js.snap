// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback component renders correctly in memory 1`] = `
<React.Fragment>
  <NavBar />
  <RoomFeedback />
</React.Fragment>
`;
